api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -368,12 +368,37 @@ public Workspace findByWorkspaceId(long workspaceId) {
       Map<String, WorkspaceAccessEntry> rolesMap) {
     List<UserRole> userRoles = new ArrayList<>();
     for (Map.Entry<String, WorkspaceAccessEntry> entry : rolesMap.entrySet()) {
-      User user = userDao.findUserByEmail(entry.getKey());
-      userRoles.add(workspaceMapper.toApiUserRole(user, entry.getValue()));
+      // Filter out groups
+      try {
+        User user = userDao.findUserByEmail(entry.getKey());
+        userRoles.add(workspaceMapper.toApiUserRole(user, entry.getValue()));
+      } catch (NullPointerException e) {
```

Comment:
Please don't catch NPEs, just do an if check on the `null` user if that's what this is checking